cdc/owner/ddl_sink.go

```diff
@@ -352,13 +352,14 @@ func (s *ddlSinkImpl) isInitialized() bool {
 }
 
 // addSpecialComment translate tidb feature to comment
-func addSpecialComment(ddlQuery string) (string, error) {
-	stms, _, err := parser.New().ParseSQL(ddlQuery)
+func addSpecialComment(ddl *model.DDLEvent) (string, error) {
+	stms, _, err := parser.New().Parse(ddl.Query, ddl.Charset, ddl.Collate)
```

Comment:
What if `ddl.Charset` is nil?

Reply:
It may happen if the TiDB is in the old version compared to the TiCDC, so the charset is "", so it's ignored.

Reply:
It's the default charset and collate, `utf8mb4`